test/mason/spec-parser/test-spec-parser.good

```diff
@@ -59,14 +59,14 @@ Spec: ncurses@6.0.0%clang@9.0.0-apple patches=f84b2708a42777aadcc7f502a261afe10c
 Package: ncurses
 Version: 6.0.0
 Compiler: ----
-Variants: patches=f84b2708a42777aadcc7f502a261afe10ca5646a51c1ef8b5e60d2070d926b57 ~symlinks arch=darwin-sierra-x86_64 ^pkg @0.29.2 %clang @9.0.0 +internal_glib arch=darwin-sierra-x86_64
+Variants: patches=f84b2708a42777aadcc7f502a261afe10ca5646a51c1ef8b5e60d2070d926b57 ~symlinks arch=darwin-sierra-x86_64 ^pkg -config@0.29.2 %clang@9.0.0-apple +internal_glib arch=darwin-sierra-x86_64
```

Comment:
note: the updated `.good` is not really the desired tokenization for variants, but neither is the original `.good`. Variants are currently unused in `mason external`, so this is not an important change.